File: easy/186_max_range_sum.py
import sys

for line in open(sys.argv[1]):
    n_days, data = line.split(';')
    n_days = int(n_days)
    data = [int(x) for x in data.split()]

#    m = max([sum(data[i:i+n_days]) for i in range(len(data) - n_days + 1)])
#    print(m if m > 0 else 0)

#    totals = []
#    for i in range(0, len(data) - n_days + 1):
#        totals.append(sum(data[i:i+n_days]))
#    m = max(totals)
#    print m if m > 0 else 0

    max_ = sum(data[0:n_days])
    total = max_
    for i in range(1, len(data) - n_days + 1):
        total -= data[i-1]
        total += data[i+n_days-1]
        if total > max_:
            max_ = total
    print(max(max_, 0))

File: easy/186_max_range_sum.c
#include <stdio.h>
#include <strings.h>
#include <string.h>
#include <stdlib.h>

#define MAX_NUMBERS 100

int main(int argc, const char * argv[]) {
    FILE *file = fopen(argv[1], "r");
    char line[1024];
    char *token;
    int n_days;
    int data[MAX_NUMBERS];
    int data_length;
    int max, total;

    while (fgets(line, 1024, file)) {
        n_days = atoi(line);

        for (data_length=0, token=strtok(index(line, ';') + 1, " ");
             token!=NULL;
             token=strtok(NULL, " "), data_length++) {
            data[data_length] = atoi(token);
        }

        max = 0;
        for (int i=0; i<n_days; i++)
            max += data[i];
        total = max;

        for (int i=1; (i+n_days-1)<data_length; i++) {
            total -= data[i-1];
            total += data[i+n_days-1];
            if (total > max)
                max = total;
        }

        printf("%d\n", (max > 0) ? max : 0);
    }

    return 0;
}
